src/core/ColorUtil.h

```diff
@@ -29,3 +29,15 @@ Color4f getContrastColor(const Color4f& col);
 Color4f getColorHSV(const Color4f& col);
 
 } // namespace OpenSCAD
+
+namespace {
+
+void rgbtohsv(float r, float g, float b, float& h, float& s, float& v);
+
+std::optional<Color4f> parse_hex_color(const std::string& hex);
+
+std::optional<Color4f> parse_web_color(const std::string& col);
+
+extern std::unordered_map<std::string, Color4f> webcolors;
+
+} // namespace
```

Comment:
Almost: Remove this from the header, then wrap the corresponding implementations in an anonymous namespace in the .cc file.

Reply:
Sorry, this is the largest C++ project I've worked on.